### ISSUE TITLE
"Error: Cannot encode [object Object]to a Firestore Value. Local testing does not yet support Firestore geo points" when using DocReference/Timestamp

### ISSUE THREAD

**Issue by ChromeQ** [firebase/firebase-functions-test#101]
This is a very similar problem raised #17 ([Error: Cannot encode [object Object]to a Firestore Value. Local testing does not yet support Firestore geo points.](https://github.com/firebase/firebase-functions-test/issues/47)) here but I am not using GeoPoints, I am adding a document reference to another document.

Also happens to document containing a firestore Timestamp

### Version info
firebase-functions-test: 0.3.0
firebase-functions: 3.14.1
firebase-admin: NOT REQUIRED
@firebase/testing: 0.20.11

### Test case
```ts
import * as firebase from '@firebase/testing';
import * as functions from 'firebase-functions';
import functionsTest from 'firebase-functions-test';

const app = firebase.initializeAdminApp({ projectId: 'my-project' });
const firestore = app.firestore();
const testEnv = functionsTest();

// example function trigger
const userCreated = functions.firestore.document('/users/{userID}').onCreate((snap) => {
    console.log(`user created: ${snap.id}!`);
});

// wrap the function
const wrapped = testEnv.wrap(userCreated);

it('test fails', async () => {
    // create a user and store the reference
    await firestore.collection('users').doc('user1').set({ name: 'Mrs. Jones' });
    const ref = firestore.collection('users').doc('user1');

    // create new user linked to the reference above
    const newUser = {
        name: 'Tom Jones',
        spouse: ref,
    };

    // try to create a snapshot of new user to pass to userCreated function
    const snap = testEnv.firestore.makeDocumentSnapshot(newUser, '/users/user2');
    const prom = wrapped(snap);

    prom.then(() => {
        return console.log('complete');
    }).catch((error: unknown) => {
        return console.error(error);
    });
});
```

### Steps to reproduce
Execute test case provided, error is thrown during the call to test.firestore.makeDocumentSnapshot.

### Expected behavior
Snapshot created successfully, function is execution is emulated as is trigger has occurred.

### Actual behavior
Call to test.firestore.makeDocumentSnapshot fails due to GeoPoint in the data model.

### Produces error:
`Cannot encode [object Object]to a Firestore Value. Local testing does not yet support Firestore geo points.`

**Comment by ChromeQ** [comment]
I've stepped into the code and found the problem seems to arise here:
https://github.com/firebase/firebase-functions-test/blob/master/src/providers/firestore.ts#L185
For some reason the document I pass in is not an instanceof firestore.DocumentReference (firestore imported from 'firebase-admin';)

So, I changed my test to import firebase and initialize firestore using firestore-admin:
```ts
import * as firebase from 'firebase-admin';
import * as functions from 'firebase-functions';
import functionsTest from 'firebase-functions-test';

const app = firebase.initializeApp({ projectId: 'last-man-standing-chromeq-dev' });
const firestore = app.firestore();
const testEnv = functionsTest();
```

But the same result:
`Cannot encode [object Object]to a Firestore Value. Local testing does not yet support Firestore geo points.`

**Comment by ChromeQ** [comment]
Finally, I found a stack overflow with a similar issue: https://stackoverflow.com/questions/58867153/timestamp-not-working-with-firebase-functions-test

The answer suggests to create a `new DocumentReference` which makes sense to pass the `if (val instanceof firestore.DocumentReference) {` but I cannot seem to do that (typescript complaints) and seems like not the most common way to create a document reference inside a document anyway.

`Constructor of class 'DocumentReference<T>' is private and only accessible within the class declaration.ts(2673)`
![image](https://user-images.githubusercontent.com/625203/120965434-3b677080-c7a8-11eb-8f17-bd79451d5234.png)

**Comment by i14h** [comment]
thanks @ChromeQ for looking into this. @joehan can you take a look at this?

**Comment by inlined** [comment]
You don't create a `DocumentReference` with the `new` operator. You create one with `firebase.firestore.document('foo/bar')`

**Comment by ChromeQ** [comment]
Yes that is the point I was making, to be able to pass the `instanceof firestore.DocumentReference` check you'd need to call `new Documentreference` which is not the correct way.

But looks as though this is fixed in v0.3.1 and #47 is closed by #105
